src/app/settings/action-triggers/components/action-triggers/action-triggers.component.ts

```diff
@@ -0,0 +1,100 @@
+  public selectedAction?: ActionTriggerWithAction<KeyboardTriggerData>
+  public createAction: boolean
+  public loading: boolean = true
+  public actions: Tv2PartAction[]
+  public actionTriggersWithAction: ActionTriggerWithAction<KeyboardTriggerData>[]
+  private readonly unsubscribe$: Subject<null> = new Subject<null>()
+
+  constructor(
+    private readonly actionTriggerStateService: ActionTriggerStateService,
+    private readonly fileDownloadService: FileDownloadService,
+    private readonly logger: Logger,
+    private readonly actionStateService: ActionStateService
+  ) {}
+
+  public ngOnInit(): void {
+    this.actionStateService
+      .subscribeToRundownActions()
+      .then(observable => observable.subscribe(actions => this.onActionsChanged(actions as Tv2PartAction[])))
+      .then(() => this.subscribeForActionTriggerObservable())
+      .catch(error => this.logger.data(error).error('Error while listening to Action events'))
+  }
+
+  private subscribeForActionTriggerObservable(): void {
+    this.actionTriggerStateService
+      .getActionTriggerObservable()
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe(actionTriggers => {
+        this.actionTriggersWithAction = JSON.parse(
+          JSON.stringify(
+            (actionTriggers as ActionTrigger<KeyboardTriggerData>[]).map(actionTrigger => {
+              return { actionTrigger: { ...actionTrigger }, action: this.actions.find(action => action.id === actionTrigger.actionId) }
+            })
+          )
+        )
+        this.setIsLoading(false)
+      })
+  }
+
+  private onActionsChanged(loadedActions: Tv2PartAction[]): void {
+    this.actions = JSON.parse(JSON.stringify(loadedActions))
+  }
+
+  public ngOnDestroy(): void {
+    this.unsubscribe$.next(null)
+    this.unsubscribe$.unsubscribe()
+  }
+
+  public setIsLoading(isLoading: boolean): void {
+    this.loading = isLoading
+  }
+
+  public selectActionTriggerForEditing(selectedTrigger?: ActionTriggerWithAction<KeyboardTriggerData>): void {
+    this.selectedAction = selectedTrigger
+    this.createAction = false
+  }
+
+  public createActionTrigger(): void {
+    this.createAction = true
+    this.selectedAction = undefined
+  }
+
+  public cancelActionTrigger(): void {
+    this.selectedAction = undefined
+    this.createAction = false
+  }
+
+  public exportActionsTriggers(): void {
+    const triggersCopy: ActionTrigger<KeyboardTriggerData>[] = this.actionTriggersWithAction.map(actionTriggerWithActions => {
+      const actionTrigger = actionTriggerWithActions.actionTrigger
+      return {
+        actionId: actionTrigger.actionId,
+        id: actionTrigger.id,
+        data: {
+          keys: actionTrigger.data.keys,
+          actionArguments: actionTrigger.data.actionArguments,
+          label: actionTrigger.data.label,
+          triggerOn: actionTrigger.data.triggerOn,
+          mappedToKeys: actionTrigger.data.mappedToKeys,
+        },
+      }
```

Comment:
Is the re-definition of the object necessary?
And if so, can't it be simplified to use the spread operator?